runtime/executor/src/main/java/edu/snu/coral/runtime/executor/bytetransfer/ContextManager.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2018 Seoul National University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.coral.runtime.executor.bytetransfer;
+
+import edu.snu.coral.runtime.common.comm.ControlMessage.ByteTransferContextSetupMessage;
+import edu.snu.coral.runtime.common.comm.ControlMessage.ByteTransferDataDirection;
+import edu.snu.coral.runtime.executor.bytetransfer.ByteTransferContext.ContextId;
+import edu.snu.coral.runtime.executor.data.BlockManagerWorker;
+import io.netty.channel.*;
+import io.netty.channel.group.ChannelGroup;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+/**
+ * Manages multiple transport contexts for one channel.
+ */
+final class ContextManager extends SimpleChannelInboundHandler<ByteTransferContextSetupMessage> {
+
+  private final BlockManagerWorker blockManagerWorker;
+  private final ByteTransfer byteTransfer;
+  private final ChannelGroup channelGroup;
+  private final String localExecutorId;
+  private final Channel channel;
+  private volatile String remoteExecutorId = null;
+
+  private final ConcurrentMap<Integer, ByteInputContext> inputContextsInitiatedByLocal = new ConcurrentHashMap<>();
```

Comment:
Why do we use `Map` between index and object? (Why not `List`?)

Reply:
Two reasons:
- To detect (potential) transferIndex conflict
- Used by `ContextManager#getInputContext` method to provide incoming data on the corresponding context.